docs/selectors.md

```diff
@@ -84,9 +84,45 @@ Id engines are selecting based on the corresponding atrribute value. For example
 
 ### zs
 
-ZSelector is an experimental engine that tries to make selectors survive future refactorings. Example: `zs=div ~ "Login"`.
+Z-selector is an experimental engine to define selectors that can **survive future refactorings**. Changes in the UI markup can often cause existing UI selectors to break. While `data-*` attributes can be useful to establish contracts between app and test code, they can be an additional maintenance burden. It can also be difficult to introduce new attributes if you are testing third-party UI.
```

Comment:
Based on the description below, it seems like these selectors can only survive certain types of refactorings. So I wonder if we should be a bit clearer here?

Reply:
> While `data-*` attributes can be useful to establish 

This remains a preferred way of testing, I'd rather say that `data-*` attributes are not always there when you start testing.

Reply:
updated to reflect this